include/Value.h

```diff
@@ -54,27 +54,25 @@ class large_value_type {
 	}
 
 public:
-	bool operator==(const large_value_type &rhs) const { return memcmp(value_, rhs.value_, sizeof(T)) == 0; }
-	bool operator!=(const large_value_type &rhs) const { return memcmp(value_, rhs.value_, sizeof(T)) != 0; }
+	bool operator==(const value_type_large &rhs) const { return memcmp(value_, rhs.value_, sizeof(T)) == 0; }
+	bool operator!=(const value_type_large &rhs) const { return memcmp(value_, rhs.value_, sizeof(T)) != 0; }
 
 public:
 	QString toHexString() const {
-		char buf[1024];
+		char buf[sizeof(T) * 2 + 1];
 		char *p = buf;
-		auto ptr = reinterpret_cast<const char *>(value_);
-		auto end = ptr + sizeof(T);
 
-		for(auto it = end; it != (ptr - 1); --it) {
-			p += sprintf(p, "%02x", *it & 0xff);
+		for(auto it = std::rbegin(value_); it != std::rend(value_); ++it) {
+			p += sprintf(p, "%016llx", *it & 0xffffffffffffffffULL);
```

Comment:
Why do you need this `& 0xffffffffffffffffULL` now? It's a NOP. Besides, such a form of a number is not really readable: why not write e.g. `0xffff'ffff'ffff'ffffULL` taking advantage of C++14 digit separators feature? (or even `uint64_t(-1)`, to make the intent explicit)

Reply:
it was left over from when I did it bytewise, you are right, isn't needed at all!